.github/workflows/build.yaml

```
@@ -19,3 +19,12 @@ jobs:
       - name: Build with Maven
         run: mvn --batch-mode --update-snapshots verify
 
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          directory: ./miranum-coverage/target/coverage/
```

```suggestion
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: ./miranum-coverage/target/coverage/
```

Looking good. But you should add the codecov token. In the past we had from time to time connection issues with codecov if no token was present

---

Ahh that's good to know. Their documentation actually recommended not to use a token for public repositories.

---

Yes, we actually did that and had rate limiting and connection issues which caused our pipelines to fail randomly. Therefore, we added the token 😄